test/httpClient.test.ts

```diff
@@ -0,0 +1,26 @@
+import http from "node:http";
+
+import { integrationTest, readAppmap, spawnAppmapNode } from "./helpers";
+import { SERVER_PORT, TEST_HEADER_VALUE } from "./httpClient";
+
+integrationTest("mapping http client requests", async () => {
+  const server = http
+    .createServer(function (req, res) {
+      if (req.url?.startsWith("/endpoint/one") ?? req.url?.startsWith("endpoint/two")) {
+        res.write("Hello World!");
+        res.end();
+      } else {
+        res.writeHead(404, { "Content-Type": "text/html" });
+        res.end();
+      }
+    })
+    .listen(SERVER_PORT);
+  const client = spawnAppmapNode("yarn", "exec", "ts-node", "index.ts");
+  await new Promise<void>((r) => client.on("close", () => r()));
```

Comment:
Note you could have just passed `r` directly here instead of wrapping it in a lambda.